From f824d5db7b45addefaa2d765a7c8b24472969c00 Mon Sep 17 00:00:00 2001
From: Bryan O'Donoghue <bryan.odonoghue@linaro.org>
Date: Wed, 28 Jun 2023 20:35:33 +0100
Subject: [PATCH 110/173] arm64: dts: qcom: sc8280xp-x13s: Define the IR sensor
 on cci1_i2c0

The 9734 is at i2c address 0x36 located on pins
cci_i2c_sda2/cci_i2c_scl2. MCLK4 is used to clock the chip with vreg_l3q,
vreg_l5q and vreg_l1q providing dovdd, avdd and dvdd respectively.

Signed-off-by: Bryan O'Donoghue <bryan.odonoghue@linaro.org>
---
 .../qcom/sc8280xp-lenovo-thinkpad-x13s.dts    | 51 +++++++++++++++++++
 1 file changed, 51 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/sc8280xp-lenovo-thinkpad-x13s.dts b/arch/arm64/boot/dts/qcom/sc8280xp-lenovo-thinkpad-x13s.dts
index 4816e2034ad6..d7895c3170b3 100644
--- a/arch/arm64/boot/dts/qcom/sc8280xp-lenovo-thinkpad-x13s.dts
+++ b/arch/arm64/boot/dts/qcom/sc8280xp-lenovo-thinkpad-x13s.dts
@@ -593,6 +593,57 @@ csiphy0_lanes01_ep: endpoint@0 {
 				data-lanes = <0 1>;
 				remote-endpoint = <&ov5675_ep>;
 			};
+
+			csiphy0_lane2_ep: endpoint@1 {
+				reg = <1>;
+				clock-lanes = <7>;
+				data-lanes = <2>;
+				remote-endpoint = <&ov9734_ep>;
+			};
+		};
+	};
+};
+
+&cci1 {
+	status = "okay";
+};
+
+&cci1_i2c0 {
+	/*
+	 * CCI_I2C2_SDA
+	 * CCI_I2C2_SCL
+	 * CAM_MCLK4
+	 * CAMERA_A0_CLK_M
+	 * CAMERA_NC_CLK_P
+	 * CAMERA_B1_LN1_M
+	 * CAMERA_A1_LN1_P
+	 * CAMERA_C0_LN0_M
+	 * CAMERA_B0_LN0_P
+	 */
+	camera@36 {
+		compatible = "ovti,ov9734";
+		reg = <0x36>;
+
+		reset-gpios = <&tlmm 7 GPIO_ACTIVE_LOW>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&cam_ir_default>;
+
+		clocks = <&camcc CAM_CC_MCLK4_CLK>;
+		clock-frequency = <19200000>;
+
+		orientation = <0>;	/* Front facing */
+
+		avdd-supply = <&pm8008_l5>;
+		dvdd-supply = <&pm8008_l1>;
+		dovdd-supply = <&pm8008_l3>;
+
+		port {
+			ov9734_ep: endpoint {
+				clock-lanes = <0>;
+				data-lanes = <1>;
+				link-frequencies = /bits/ 64 <180000000>;
+				remote-endpoint = <&csiphy0_lane2_ep>;
+			};
 		};
 	};
 };
-- 
2.43.0

